### ISSUE TITLE
Update project visibility on permissions

### ISSUE THREAD

**Issue by sra405** [RaspberryPiFoundation/editor-api#394]
We need to update the API based on user permissions to enable teachers/owners to view students projects but ensure students can't see each others work. Various API thoughts were initially discussed [here](https://docs.google.com/document/d/1VC3FDOHMd2tiLOte6QQCJIS4yy_MR8DAxxHsI8wTKAY/edit?usp=sharing) (see below)

Project visibility permissions - how can we limit what projects students can view?
- We need them to see their teachers projects but currently have school and lesson IDs
  - Only give the teachers project the lesson ID
- How do we ensure they don’t have access to other students projects
  - Student projects won’t have the lesson ID assigned
- Ensure that teachers can access their students’ work
  - Use the remixed from ID to give the teacher access to their student work
  - Every Classroom Management project is given the school ID

In summary:
  - [ ] Students create projects without a lesson ID (only ever remix)
  - [ ] Student read for their work should just work (same as projects integration auto load remix)
  - [x] Students needs access to teachers/lesson project (this currently works but the hide/show functionality doesn't)
  - [x] Teacher needs access to students work (from remixed ID)
  - [x] Check remix doesn’t copy lesson ID but does copy school ID (i.e. teachers/owners see it based off the remix ID but other students don't because it doesn't have a lesson ID)

**Comment by loiswells97** [comment]
Do we need to change the visibility of the code project to students based on the visibility of the lesson as well? Or do we already have this?
